doc/source/user_guide/copy_on_write.rst

```diff
@@ -0,0 +1,183 @@
+.. _copy_on_write:
+
+{{ header }}
+
+*******************
+Copy-on-Write (CoW)
+*******************
+
+.. ipython:: python
+    :suppress:
+
+    pd.options.mode.copy_on_write = True
+
+Copy-on-Write was first introduced in version 1.5.0. Starting from version 2.0 most of the
+optimizations that become possible through CoW are implemented and supported. A complete list
+can be found at TODO
+
+We expect that CoW will be enabled per default in version 3.0
+
+Description
+-----------
+
+CoW means that any DataFrame or Series derived from another in any way always
+behaves as a copy. As a consequence, we can only change the values of an object
+through modifying the object itself. CoW disallows updating a DataFrame or a Series
```

Comment:
I think it would be valuable to show a before/after code example of CoW where modifying a derived result modifies the parent/doesn't modify the parent respectively

Reply:
I'll make a specific section for this. Since reset_index copies right now, this does not fit in here very well.